### PR TITLE
Add destinations to data source interface

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Alters the `DataSource` interface so that it has 3 potential destinations where it can write data: spans, span events, and logs.

This ties each `DataSource` to capturing only one type of data directly (Span/SpanEvent/Logs). IMO this is preferable and in edge-cases it should be possible to create multiple implementations.

## Testing

Added unit test coverage.

### CODE REVIEW

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/DataSource.kt [hunk 1]
```diff
@@ -16,3 +30,24 @@ internal interface DataSource {
...
+
+/**
+ * A [DataSource] that adds a new log to the log service.
+ */
+internal interface LogDataSource : DataSource<LogService>
+
+/**
+ * Placeholder type for the log service.
+ */
+internal interface LogService
```
Comment (fractalwrench) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/DataSource.kt:53]: This is just a placeholder type for now, as we haven't defined the contract for what log capture should look like.

**Review by bidetofevil — APPROVED** [review]
Lets give this a go to see if we need more specificity
